src/components/text-editor/link-menu/editor-link-menu.tsx

```diff
@@ -122,14 +142,32 @@ export class TextEditorLinkMenu {
         return !!getHref(href);
     };
 
+    private handleKeyDown = (event: KeyboardEvent) => {
+        if (event.key !== ENTER) {
+            return;
+        }
+
+        if (this.saveButton) {
+            this.saveButton.focus();
+        }
+
+        event.preventDefault();
+        if (this.link?.href && !this.invalidLink) {
+            this.handleSave(event);
+        }
+    };
+
     private handleCancel = (event: MouseEvent) => {
         this.cancel.emit();
         event.stopPropagation();
     };
 
     private handleSave = (event: MouseEvent | KeyboardEvent) => {
-        this.save.emit();
         event.stopPropagation();
+
+        requestAnimationFrame(() => {
+            this.save.emit();
+        });
```

Comment:
Again, why do we have requestAnimationFrame here?

Reply:
This is a leftover from when we were trying to fix the issue that turned out to be caused by the debounce in `limel-input-field`. I was trying to fix it without adding a debounce to the save emit function. I forgot to remove it and it can be removed from here.